network.go

```diff
@@ -135,3 +152,34 @@ func getIPFromHeader(r *http.Request, name string) net.IP {
 
 	return nil
 }
+
+func extractSubFromJWT(jwtString, secret string) (string, error) {
+	claims := &jwt.StandardClaims{}
+	token, err := jwt.ParseWithClaims(jwtString, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(secret), nil
+	})
+	if err != nil {
+		return "", err
+	}
+	if !token.Valid {
+		return "", http.ErrLineTooLong
+	}
+	fmt.Println(claims.Subject)
+	return fmt.Sprint([]byte(claims.Subject)), nil
+}
+
+func getAuthorizationToken(r *http.Request) string {
```

Comment:
Please returns a boolean to explicitly communicate if your function was able to extract authorization token.

```suggestion
func getAuthorizationToken(r *http.Request) (string, bool) {
```